sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/HttpClientProviders.java

```diff
@@ -56,14 +67,21 @@ public static HttpClient createInstance() {
     }
 
     public static HttpClient createInstance(ClientOptions clientOptions) {
-        if (defaultProvider == null) {
+        if (DEFAULT_PROVIDER == null) {
             throw LOGGER.logExceptionAsError(new IllegalStateException(CANNOT_FIND_HTTP_CLIENT));
         }
 
         if (clientOptions instanceof HttpClientOptions) {
-            return defaultProvider.createInstance((HttpClientOptions) clientOptions);
+            HttpClientOptions httpClientOptions = (HttpClientOptions) clientOptions;
+            String selectedImplementation = httpClientOptions.getHttpClientImplementation();
+            if (CoreUtils.isNullOrEmpty(selectedImplementation)) {
+                return DEFAULT_PROVIDER.createInstance(httpClientOptions);
+            } else {
+                AVAILABLE_PROVIDERS.getOrDefault(selectedImplementation, DEFAULT_PROVIDER)
```

Comment:
if user explicitly mentioned they wanted okhttp, but it's not found on the classpath we should throw rather than return default.